trunk/src/app/srs_app_source.cpp

```diff
@@ -855,6 +855,13 @@ srs_error_t SrsOriginHub::initialize(SrsLiveSource* s, SrsRequest* r)
     req_ = r;
     source = s;
     
+    int port; string ip;
+    std::vector<std::string> listens = _srs_config->get_listens();
+    if(listens.size() > 0){
+	srs_parse_endpoint(listens[0], ip, port);
+	req_->port = port;
```

Comment:
It seems that it's caused by the transcoding variable substitution, so we should fix the transcoding.

We MUST NOT change the port to listen port in Request, because the port is request.

Does that make sense?